### PR TITLE
Support for Code Snippets in Editor

### PR DESCRIPTION
**Pull request by julian-r**

### CODE REVIEW

File: frontend/src/Monaco.tsx [hunk 1]
```diff
@@ -64,29 +71,72 @@ function Monaco({ args }: AceProps) {
       throw new Error("Container is not available");
     }
 
-    setDiagnosticsOptions({
+    console.log("Monaco useEffect called")
+    console.log(currentArgs.current.snippets);
```
Comment (trojanowski) [inline frontend/src/Monaco.tsx:75]: To remove

File: frontend/src/Monaco.tsx [hunk 2]
```diff
@@ -64,29 +71,72 @@ function Monaco({ args }: AceProps) {
...
+
+    // Completion for code snippets
+    monaco.languages.registerCompletionItemProvider('yaml', {
+      provideCompletionItems: function (
+        model: monaco.editor.ITextModel,
+        position: monaco.Position,
+        context: monaco.languages.CompletionContext,
+        token: monaco.CancellationToken) {
+
+        console.log("called completion")
```
Comment (trojanowski) [inline frontend/src/Monaco.tsx:121]: To remove

**Review by trojanowski — CHANGES_REQUESTED** [review]
@julian-r currently we should also run `./build.sh` and add the changes to the commit. Alternatively we could setup CI to do it automatically there.

File: frontend/src/Monaco.tsx [hunk 3]
```diff
@@ -64,29 +71,72 @@ function Monaco({ args }: AceProps) {
...
+    function cerateSnippets(range: monaco.IRange): monaco.languages.CompletionItem[] {
+      return currentArgs.current.snippets.map((snippet: any): monaco.languages.CompletionItem => {
+
+        return {
+          label: snippet.label,
+          kind: monaco.languages.CompletionItemKind.Enum,
+          insertText: snippet.insertText,
+          range: range,
+          detail: snippet.detail,
+          // documentation: snippet.detail,
```
Comment (trojanowski) [inline frontend/src/Monaco.tsx:86]: To delete?

File: streamlit_monaco_yaml/__init__.py [hunk 4]
```diff
@@ -17,8 +17,31 @@
...
+    label: str
+    insertText: str
+    detail: str
+
+
 def monaco_editor(
-    value: str = "", schema: Optional[dict] = None, height: int = 600, key: str = None
+    value: str = "",
+    schema: Optional[dict] = None,
+    snippets: list[Snippet] = None,
```
[the remark this answers is not in the document]
Comment (julian-r) [reply streamlit_monaco_yaml/__init__.py:36]: A list is expected :)

**Review by trojanowski — APPROVED** [review]

**Pull request merged**